job-service/src/main/java/com/hpe/caf/services/job/api/DatabaseHelper.java

```diff
@@ -403,7 +415,15 @@ public boolean canJobBeProgressed(final String partitionId, final String jobId)
             }
 
         } catch (final SQLException se) {
-            throw mapSqlException(se);
+            final String sqlState = se.getSQLState();
+            if (sqlState.equals(POSTGRES_NO_DATA_FOUND_ERROR_CODE)) {
+                // job missing - return false
+            } else if (sqlState.startsWith(POSTGRES_CONNECTION_EXCEPTION_ERROR_CODE_PREFIX)) {
+                // Connection exception
+                throw new ServiceUnavailableException(se.getMessage());
+            } else {
+                throw se;
+            }
```

Comment:
It is not possible to create a method to encapsulate this block because the new method will complain the missing return statement from sqlState.equals(POSTGRES_NO_DATA_FOUND_ERROR_CODE)

Reply:
You can encapsulate:

```
final String sqlState = se.getSQLState();
if (sqlState.equals(POSTGRES_NO_DATA_FOUND_ERROR_CODE)) {
   // job missing - return false
} else if (sqlState.startsWith(POSTGRES_CONNECTION_EXCEPTION_ERROR_CODE_PREFIX)) {
   // Connection exception
   throw new ServiceUnavailableException(se.getMessage());
}  else {
   throw e;	
}

```
into a method that looks like this:

```
private static void throwIfUnexpectedException(final SqlException sqlException) throws Exception {
    final String sqlState = se.getSQLState();
    if (sqlState.equals(POSTGRES_NO_DATA_FOUND_ERROR_CODE)) {
       // job missing - return false
    } else if (sqlState.startsWith(POSTGRES_CONNECTION_EXCEPTION_ERROR_CODE_PREFIX)) {
       // Connection exception
       throw new ServiceUnavailableException(se.getMessage());
    }  else {
       throw e;	
    }
}
```